src/main/java/com/hivemq/mqtt/handler/connect/MessageBarrier.java

```diff
@@ -54,8 +50,8 @@ public class MessageBarrier extends ChannelDuplexHandler {
     private final @NotNull MqttServerDisconnector serverDisconnector;
     private final @NotNull Queue<Message> messageQueue = new LinkedList<>();
 
-    private boolean connectReceived = false;
-    private boolean connackSent = false;
+    private boolean connectReceived;
+    private boolean connackSent;
```

Comment:
will the above two be set to false by default? IMHO, even if they are, it makes sense to make it explicit in the ctor

Reply:
Yes, Java will initialize boolean per default with false.